app/ide-desktop/lib/dashboard/src/authentication/cognito.ts

```diff
@@ -367,19 +371,49 @@ export interface UserSession {
   readonly email: string
   /** User's access token, used to authenticate the user (e.g., when making API calls). */
   readonly accessToken: string
+  /** User's refresh token, used to refresh the access token when it expires. */
+  readonly refreshToken: string
+  /** URL to refresh the access token. */
+  readonly refreshUrl: string
+  /** Time when the access token will expire, date in ISO format(UTC). */
+  readonly expireAt: string
+  /** The client ID of the user pool. */
+  readonly clientId: config.UserPoolWebClientId
 }
 
 /** Parse a `CognitoUserSession` into a {@link UserSession}.
  * @throws If the `email` field of the payload is not a string. */
-function parseUserSession(session: cognito.CognitoUserSession): UserSession {
+function parseUserSession(
+  session: cognito.CognitoUserSession,
+  clientId: config.UserPoolWebClientId
+): UserSession {
   const payload: Readonly<Record<string, unknown>> = session.getIdToken().payload
   const email = payload.email
+
   /** The `email` field is mandatory, so we assert that it exists and is a string. */
   if (typeof email !== 'string') {
     throw new Error('Payload does not have an email field.')
   } else {
     const accessToken = session.getAccessToken().getJwtToken()
-    return { email, accessToken }
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const refreshUrl: string = session.getAccessToken().payload.iss
+    const expirationTimestamp = session.getAccessToken().getExpiration()
+
+    const expireAt = (() => {
+      const date = new Date(0)
+      date.setUTCSeconds(expirationTimestamp)
```

Comment:
what's the difference between this and `new Date(expirationTimestamp * 1_000)` btw?

Reply:
The result is the same, just used to write that way. Not sure which solution is better though.

Reply:
probably subjective, the main differences would be the added line and the function call statement (as opposed to a function call expression)